vendor/bat-native-ads/src/bat/ads/internal/ads/serving/choose/sample_ads.h

```diff
@@ -16,17 +17,28 @@
 
 namespace ads {
 
+template <typename T>
+T CalculateNormalizingConstantBase(const std::vector<T>& scores) {
+  T normalizing_constant = 0;
+  for (const auto& score : scores) {
+    normalizing_constant += score;
+  }
+
+  return normalizing_constant;
+}
+
 template <typename T>
 double CalculateNormalizingConstant(
     const CreativeAdPredictorMap<T>& creative_ad_predictors) {
-  double normalizing_constant = 0.0;
+  std::vector<double> scores;
+  scores.assign(creative_ad_predictors.size(), 0);
```

Comment:
Would this not break the v2 ad serving pipeline? Because we need to be backwards compatible